Tests/FuzzilliTests/ProgramBuilderTest.swift

```diff
@@ -251,6 +251,392 @@ class ProgramBuilderTests: XCTestCase {
+                let v0 = b.loadInt(42)
+                let _ = b.createObject(with: ["foo": v0])
+                b.await(value: v3)
+                let v8 = b.loadInt(1337)
+                b.yield(value: v8)
+            }
+            b.doReturn(value: v4)
+        }
+
+        let original = b.finalize()
+
+        b.defineAsyncFunction(withSignature: FunctionSignature(withParameterCount: 2)) { _ in
+            // Splicing at Await
+            b.splice(from: original, at: original.code.lastInstruction.index - 5)
+        }
+        
+        let actualSplice = b.finalize()
+
+        b.defineAsyncFunction(withSignature: FunctionSignature(withParameterCount: 2)) { _ in
+            let v0 = b.loadInt(0)
+            let _ = b.await(value: v0)
+        }
+        let expectedSplice = b.finalize()
+
+        XCTAssertEqual(actualSplice, expectedSplice)
+    }
+
+    func testForInSplicing() {
+        let fuzzer = makeMockFuzzer()
+        let b = fuzzer.makeBuilder()
+        b.mode = .conservative
+        
+        b.defineAsyncFunction(withSignature: FunctionSignature(withParameterCount: 2)) { _ in
+
+        //BEGIN: Some instructions that shouldn't end up in the splice
+        var initialValues = [Variable]()
+        initialValues.append(b.loadInt(15))
+        initialValues.append(b.loadInt(30))
+        initialValues.append(b.loadString("Hello"))
+        initialValues.append(b.loadString("World"))
+        let v4 = b.createArray(with: initialValues)
+        let v5 = b.loadFloat(13.37)
+
+        let v6 = b.loadBuiltin("Array")
+        let _ = b.construct(v6, withArgs: [v4,v5], spreading: [true,false])
+        // END
+
+        let v0 = b.loadInt(10)
+        let v1 = b.loadString("Hello")
+        let v2 = b.loadFloat(13.57)
+        let v3 = b.createObject(with: ["foo": v2, "bar": v1, "baz": v0])
+        b.forInLoop(v3) { v4 in
+            let v5 = b.loadInt(1000)
+            let v6 = b.await(value: v5)
+            b.storeComputedProperty(v6, as: v4, on: v3)
+        }
+        }
+        let original = b.finalize()
+        
+        // Splicing at StoreComputedProperty
+        b.splice(from: original, at: original.code.lastInstruction.index - 2)       
+        
+        let actualSplice = b.finalize()
+
+        b.defineAsyncFunction(withSignature: FunctionSignature(withParameterCount: 2)) { _ in
+        let v0 = b.loadInt(10)
+        let v1 = b.loadString("Hello")
+        let v2 = b.loadFloat(13.57)
+        let v3 = b.createObject(with: ["foo": v2, "bar": v1, "baz": v0])
+        b.forInLoop(v3) { v4 in
+            let v5 = b.loadInt(1000)
+            let v6 = b.await(value: v5)
+            b.storeComputedProperty(v6, as: v4, on: v3)
+        }
+        }
+
+        let expectedSplice = b.finalize()
+
+        XCTAssertEqual(actualSplice, expectedSplice)
+    }
+
+    func testBeginForSplicing() {
+        let fuzzer = makeMockFuzzer()
+        let b = fuzzer.makeBuilder()
+        b.mode = .conservative
```

Comment:
Nit: I think we can remove `b.mode = .conservative` now from the tests since the splicing no longer rewires variables

Reply:
I've left this in the testcases so that we exercise the code path in the splice loop that checks if an instruction mutates any of the requiredInputs:
```
if mode == .conservative || (mode == .aggressive && probability(0.5)) {
                if instr.mayMutate(requiredInputs) {
```

Reply:
Ah cool, yeah that makes sense, maybe add a comment about that for the first test that does it, but not really important.